### PR TITLE
Add offer

### PR DESCRIPTION
**Pull request by Astenna**

### CODE REVIEW

File: src/Joblify.Core/Offers/Dto/OfferDto.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+        public double Price { get; set; }
+
+        [Required]
+        public int CategoryId { get; set; }
+
+        [Required]
+        public int TradeId { get; set; }
+
+        public string AvailableTime { get; set; }
```
Comment (Bartosz97) [inline src/Joblify.Core/Offers/Dto/OfferDto.cs:20]: AvailableTime like string? :O

File: src/Joblify.Core/Offers/OfferService.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public OfferService(IMapper mapper, IUnitOfWork unitOfWork)
+        {
+            _mapper = mapper;
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task<OfferDto> AddOfferAsync(OfferDto offerDto)
```
Comment (Bartosz97) [inline src/Joblify.Core/Offers/OfferService.cs:19]: Maybe AddOffer instead of AddOfferAsync?

File: src/Joblify.Core/Offers/IOfferService.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading.Tasks;
+
+namespace Joblify.Core.Offers
+{
+    public interface IOfferService
+    {
+        Task<OfferDto> AddOfferAsync(OfferDto osfferDto);
+
```
Comment (Bartosz97) [inline src/Joblify.Core/Offers/IOfferService.cs]: Unnecessary space

File: src/Joblify.Core.Data/Models/Offer.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        public Category Category { get; set; }
+
+        [ForeignKey("Trade")]
+        public int TradeId { get; set; }
+
+        public Trade Trade { get; set; }
+
+        public string AvailableTime { get; set; }
+
+        public bool IsDeleted { get; set; }
```
Comment (Bartosz97) [inline src/Joblify.Core.Data/Models/Offer.cs:31]: Can we set state IsDeleted like false? Or is it a bad idea? @channel

Comment (kobrynsky) [reply src/Joblify.Core.Data/Models/Offer.cs:31]: @Bartosz97  booleans are not nullable - by default they are false. (If I have understood you correctly 😄 )

Comment (Bartosz97) [reply src/Joblify.Core.Data/Models/Offer.cs:31]: I'm a pitcher. These are the effects of working for 12 hours ;C.

File: src/Joblify.Core/Offers/OfferService.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            var offer = _mapper.Map<Offer>(offerDto);
+            offer.IsDeleted = false;
```
Comment (Bartosz97) [inline src/Joblify.Core/Offers/OfferService.cs]: Is it necessarily?

File: src/Joblify.Core.Data/Models/Offer.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        public int CategoryId { get; set; }
+
+        public Category Category { get; set; }
+
+        [ForeignKey("Trade")]
+        public int TradeId { get; set; }
+
+        public Trade Trade { get; set; }
+
+        public string AvailableTime { get; set; }
```
Comment (kobrynsky) [inline src/Joblify.Core.Data/Models/Offer.cs:29]: Shouldnt AvailableTime be DateTime? 😄

File: src/Joblify.Core/Offers/OfferService.cs (hunk 2 above)
Comment (kobrynsky) [reply src/Joblify.Core/Offers/OfferService.cs:19]: @Bartosz97  Could you please explain why? 😄

File: src/Joblify.Core.Data/Models/Offer.cs (hunk 4 above)
Comment (kobrynsky) [reply src/Joblify.Core.Data/Models/Offer.cs:31]: It's all right, go get some rest C:

File: src/Joblify.Core.Data/Models/Offer.cs (hunk 6 above)
Comment (Astenna) [reply src/Joblify.Core.Data/Models/Offer.cs:29]: We should clarify what exactly that field is. Is it a time of day like 'in the afternoons' or is it a range of dates? If it's a range of dates then we should have two different fields for start and end (?)

File: src/Joblify.Core/Offers/OfferService.cs (hunk 5 above)
Comment (Astenna) [reply src/Joblify.Core/Offers/OfferService.cs]: not really, corrected c:

File: src/Joblify.Core/Offers/Dto/OfferDto.cs (hunk 1 above)
Comment (Bartosz97) [reply src/Joblify.Core/Offers/Dto/OfferDto.cs:20]: String for us is better ;)

File: src/Joblify.Core.Data/Models/Offer.cs (hunk 6 above)
Comment (Bartosz97) [reply src/Joblify.Core.Data/Models/Offer.cs:29]: String is better for us :)

**Review by Bartosz97 — APPROVED** [review]

**Review by kobrynsky — APPROVED** [review]

**Pull request merged**